hooks/airtable-conversion-utils.ts

```diff
@@ -18,7 +20,101 @@ function getThumbnailUrl (imageAttachment: any): string|null {
   return largeThumbnail ? largeThumbnail.url : null
 }
 
+class AirtableRecords {
+  protected apiKey: string;
+  private baseId: string;
+  private tableId: string;
+  private view: string;
+
+  constructor (apiKey: string, baseId: string, tableId: string, view: string) {
+    this.apiKey = apiKey
+    this.baseId = baseId
+    this.tableId = tableId
+    this.view = view
+  }
+
+  /**
+   * Get the Airtable field name for a given field ID.
+   *
+   * @param fieldId Field ID
+   * @returns {Promise<string | null>} Field name
+   */
+  public getFieldName (fieldId: string): Promise<string | null> {
+    const base = new Airtable({ apiKey: this.apiKey }).base(this.baseId)
+    let fieldName: string | undefined
+
+    try {
+      return base(this.tableId)
+        .select({
+          fields: [fieldId],
+          view: this.view
+        })
+        .eachPage((records, nextPage) => {
+          for (const record of records) {
+            if (fieldName) {
+              break
+            }
+
+            const recordFields = Object.keys(record.fields)
+
+            if (recordFields.length > 0) {
+              fieldName = recordFields[0]
+            }
+          }
+
+          nextPage()
+        })
+        .then(() => {
+          if (fieldName) {
+            return fieldName
+          } else {
+            return ''
+          }
+        })
+    } catch (error) {
+      console.error(`Error in getFieldName: ${error}`)
+      return Promise.resolve(null)
+    }
+  }
+
+  /**
+   * Get the Airtable field names for all the fields IDs provided.
+   * The object provided must be a map of keys to field IDs.
+   * The returned object will be a map of keys to field names.
+   *
+   * @param fieldIds Field IDs
+   * @returns {Promise<Record<string, string | null>>} Field names mapped to keys
+   */
+  public getAllFieldNames (
```

Comment:
So that the methods that call these methods can know that they need to wait for the names to be available, since they are the result of async operations (Airtable API calls).